apps/web/modules/event-types/views/event-types-single-view.tsx

```diff
@@ -400,40 +402,50 @@ const EventTypePage = (props: EventTypeSetupProps & { allActiveWorkflows?: Workf
     formState: { isDirty: isFormDirty, dirtyFields },
   } = formMethods;
 
-  // useEffect(() => {
-  //   const handleRouteChange = (url: string) => {
-  //     const paths = url.split("/");
-  //
-  //     // Check if event is managed event type - skip if there is assigned users
-  //     const assignedUsers = eventType.children;
-  //     const isManagedEventType = eventType.schedulingType === SchedulingType.MANAGED;
-  //     if (eventType.assignAllTeamMembers) {
-  //       return;
-  //     } else if (isManagedEventType && assignedUsers.length > 0) {
-  //       return;
-  //     }
-  //
-  //     const hosts = eventType.hosts;
-  //     if (
-  //       !leaveWithoutAssigningHosts.current &&
-  //       !!team &&
-  //       (hosts.length === 0 || assignedUsers.length === 0) &&
-  //       (url === "/event-types" || paths[1] !== "event-types")
-  //     ) {
-  //       setIsOpenAssignmentWarnDialog(true);
-  //       setPendingRoute(url);
-  //       router.events.emit(
-  //         "routeChangeError",
-  //         new Error(`Aborted route change to ${url} because none was assigned to team event`)
-  //       );
-  //       throw "Aborted";
-  //     }
-  //   };
-  //   router.events.on("routeChangeStart", handleRouteChange);
-  //   return () => {
-  //     router.events.off("routeChangeStart", handleRouteChange);
-  //   };
-  // }, [router]);
+  const initialCheck = useRef(
+    checkForEmptyAssignment({
+      assignedUsers: formMethods.getValues("children"),
+      hosts: formMethods.getValues("hosts"),
+      assignAllTeamMembers: eventType.assignAllTeamMembers,
+      isManagedEventType: eventType.schedulingType === SchedulingType.MANAGED,
+      isTeamEvent: !!team,
+    })
+  );
+
+  const assignedUsers = formMethods.watch("children");
+  const hosts = formMethods.watch("hosts");
+  const assignAllTeamMembers = formMethods.watch("assignAllTeamMembers");
+
+  useEffect(() => {
```

Comment:
As per issue we should try and extract this logic out of the useEffect into a new hook

Reply:
watch was used to make form dynamic, since we rely on Saved State only, updated.